### ISSUE TITLE
Help with Installation and configuration of OVC stack 

### ISSUE THREAD

**Issue by Mediacast-cloud** [OpenVisualCloud/Dockerfiles#503]
Hi,
We are looking for help in installing and configuring the OVC stack, in particular, the 3 repos;
1.	SVT-HEVC.
2.	WebRTC using Open RTC Kit “CWT’.
3.	CDN Transcoding Sample.

I have attached a simple requirement document to briefly outline the workflow we currently lead in which the OVC stack would fit in. If further details are needed I will be happy to provide.

[Media Server - Requirements Sheet.docx](https://github.com/OpenVisualCloud/Dockerfiles/files/4761691/Media.Server.-.Requirements.Sheet.docx)

**Comment by xwu2git** [comment]
Acknowledged. Working on your reqs to see what stacks can meet your needs.

**Comment by xwu2git** [comment]
The three projects you mentioned have their README.md for installation and configuration steps.

SVT-HEVC: https://github.com/OpenVisualCloud/SVT-HEVC/blob/master/README.md
WebRTC OWT: https://github.com/open-webrtc-toolkit/owt-server/blob/master/README.md
CDN Transcoding Sample: https://github.com/OpenVisualCloud/CDN-Transcode-Sample/README.md.

Can you highlight what are the specific issues in installation and configuration? 

Thanks

xwu2git

**Comment by Mediacast-cloud** [comment]
Thank you for your support.

the installer says he  id facing issues with the webrtc repo. `he quoted:

this is the error i am getting
scripts/build.js -t all --check
while i run this command
rtcConn.target.mk:199: recipe for target 'Release/obj.target/rtcConn/erizo/src/erizo/dtls/DtlsSocket.o'; failed
make: *** [Release/obj.target/rtcConn/erizo/src/erizo/dtls/DtlsSocket.o] Error 1
make: Leaving directory '/home/ubuntu/owt-server/source/agent/webrtc/rtcConn/build'


See attached screen shot

<img width="783" alt="Screenshot 2020-06-12 at 21 59 18" src="https://user-images.githubusercontent.com/56137879/84546266-e0b44980-acf8-11ea-94ef-2f7009c4b6f4.png">

**Comment by xwu2git** [comment]
@qwu16  will help you.

**Comment by qwu16** [comment]
There is a new commit in OWT repo which adds a new patch, so we need update related m4 file for OWT. Please check whether it fixes your issue after related pull request is committed and merged later.

**Comment by dpatel257** [comment]
> There is a new commit in OWT repo which adds a new patch, so we need update related m4 file for OWT. Please check whether it fixes your issue after related pull request is committed and merged later.

Hi @qwu16 , currently owt-server in Dockerfiles is pointing to 4.2.x branch.
Which commit are you referring to in this above comment?

**Comment by Mediacast-cloud** [comment]
Hi @qwu16 

Are you referring the active branch updated by starwarfan?

<img width="947" alt="OWT update" src="https://user-images.githubusercontent.com/56137879/84698064-18183580-af47-11ea-8220-b4b07d767b07.png">

**Comment by Mediacast-cloud** [comment]
When  trying to install from 4.3.x same error (see attached) screen shot. Also similar experience in the issue in the link below
https://www.freelancer.com/users/l.php?url=https:%2F%2Fgithub.com%2Fopen-webrtc-toolkit%2Fowt-server%2Fissues&sig=2d4ad3f167fed166f7ea5c86d088d8f20e3def70ec16fa9fa24dc4ad142dec6e

[OWT ERROR Screen 4.3x .pdf](https://github.com/OpenVisualCloud/Dockerfiles/files/4783129/OWT.ERROR.Screen.4.3x.pdf)

**Comment by qwu16** [comment]
> Hi @qwu16
> 
> Are you referring the active branch updated by starwarfan?
> 
> <img alt="OWT update" width="947" src="https://user-images.githubusercontent.com/56137879/84698064-18183580-af47-11ea-8220-b4b07d767b07.png">

Yes, this commit introduces dependency change. I have committed a PR 505 in OVC Dockerfile repo  and please try again after the PR is merged. If your issue still exists, please relocate you dependency install and OWT compile and packing log to one file so that we can better locate the issue.

**Comment by jhou5** [comment]
PR#505 is merged to master to fix owt image. I would submit another similar fix for owt-immersive image soon. @Mediacast-cloud pls. have a try and let us know if your issue still exists.

BTW, @qwu16 , for owt images, some of the components are still old version , i.e. SVT HEVC, FFMPEG. I am thinking about upgrading them through using svt-hevc.m4 and ffmpeg.m4. Do you see any side impact?

**Comment by jhou5** [comment]
The fix for owt-immersive image is also merged into master now.

**Comment by qwu16** [comment]
> PR#505 is merged to master to fix owt image. I would submit another similar fix for owt-immersive image soon. @Mediacast-cloud pls. have a try and let us know if your issue still exists.
> 
> BTW, @qwu16 , for owt images, some of the components are still old version , i.e. SVT HEVC, FFMPEG. I am thinking about upgrading them through using svt-hevc.m4 and ffmpeg.m4. Do you see any side impact?

@jhou5 Please keep current ffmpeg and SVT HEVC version for OWT. We are not sure if there is any side impact for upgrading cause we don't have a full testing with the newer version. Thanks.

**Comment by jhou5** [comment]
@qwu16 OK, I see. Thanks for the reponse.

**Comment by Mediacast-cloud** [comment]
@qwu16 
The error is still prevailing. 
Please see attached screen shots
<img width="585" alt="OWC OWT INstallation Error-2" src="https://user-images.githubusercontent.com/56137879/85011732-e4f9c000-b159-11ea-9784-4f241e6cf6c9.png">
<img width="494" alt="OVC OWT `indtall Error-1" src="https://user-images.githubusercontent.com/56137879/85011739-e6c38380-b159-11ea-906a-d825860e6eb1.png">

**Comment by qwu16** [comment]
> @qwu16
> The error is still prevailing.
> Please see attached screen shots
> <img alt="OWC OWT INstallation Error-2" width="585" src="https://user-images.githubusercontent.com/56137879/85011732-e4f9c000-b159-11ea-9784-4f241e6cf6c9.png">
> <img alt="OVC OWT `indtall Error-1" width="494" src="https://user-images.githubusercontent.com/56137879/85011739-e6c38380-b159-11ea-906a-d825860e6eb1.png">

Do you update the owt.m4 template and rebuild ovc Dockerfile? According to your snapshot, your node-gyp version is still 7.0.0, and in latest owt.m4 template, we fix the node-gyp version to 6.1.0.

**Comment by Mediacast-cloud** [comment]
Installer still facing the issue after updating the node-gyp version to 6.1.0
Could there be a dependancy issue with the node?
 
<img width="971" alt="OWT Installation error-3" src="https://user-images.githubusercontent.com/56137879/85126653-d24cbd00-b225-11ea-9857-bf1251bd2ca5.png">

**Comment by qwu16** [comment]
> Installer still facing the issue after updating the node-gyp version to 6.1.0
> Could there be a dependancy issue with the node?
> 
> <img alt="OWT Installation error-3" width="971" src="https://user-images.githubusercontent.com/56137879/85126653-d24cbd00-b225-11ea-9857-bf1251bd2ca5.png">

Are you using Dockerfiles in OVC to build OWT or you directly use OWT github master branch to install and build? Please save you dependency install and build process log into a file, then we can better locate the root cause.

**Comment by Mediacast-cloud** [comment]
Hi. The Installer is using the Master from Github.
Please, review the dependency of install and the build process log shared in the attached. I hope it helps in your diagnosis.
[build_log.log](https://github.com/OpenVisualCloud/Dockerfiles/files/4837636/build_log.log)
![Initial_build_command](https://user-images.githubusercontent.com/56137879/85871313-21ff2b80-b7c6-11ea-9e17-9162259a42d9.PNG)
![build_final_result](https://user-images.githubusercontent.com/56137879/85871319-23305880-b7c6-11ea-9549-a88fa2ba83a0.PNG)

**Comment by jhou5** [comment]
@qwu16 can you pls. help a check @Mediacast-cloud's log?

**Comment by qwu16** [comment]
@Mediacast-cloud Do you use the node 8.15.0 to build OWT? It seems that you are using a newer version 14.4.0 to build OWT, and API changed in newer version. Please change node version to OWT required 8.15.0 to have a try.

**Comment by jhou5** [comment]
@Mediacast-cloud  do you have newly update? Need more support?

**Comment by Mediacast-cloud** [comment]
Yes, @ jhou5. Our installer continues to face built issues with WebRTC and CDN installations on a Ubuntu's 18.04   OS.
Anyone can help is jump start the installation hurdle you know of?

**Comment by jhou5** [comment]
have you tried qwu16's solution?

**Comment by Mediacast-cloud** [comment]
Yes but issues were still present

**Comment by jhou5** [comment]
Just tried, I can not reproduce it on my machine. If possible, can you pls. guide me the steps?

**Comment by Mediacast-cloud** [comment]
Steps for what?

**Comment by jhou5** [comment]
For how to reproduce the failure.

**Comment by Mediacast-cloud** [comment]
I'm sorry. I am not technical but will ask the freelance installer. Do you mean you want screen shots of the issues?

**Comment by jhou5** [comment]
Not screen shots of issues. I hope to see how to reproduce the issue so as to know how to help here, as i cannot reproduce it on my side.

**Comment by Mediacast-cloud** [comment]
How can we do that? Access to the installation server?

**Comment by jhou5** [comment]
Maybe step by step for how to reproduce the issue on a clean linux OS.

**Comment by Mediacast-cloud** [comment]
Yes. I certainly agree @jhou5 .
I can prepare a fresh server tomorrow.

**Comment by Mediacast-cloud** [comment]
@jhou5  Will the below server specs suffice for installation of the OVC 3 repos (SVT-HEVC, WEbRTC and CDN)?
The VPS Server Specs are:
10 CPU cores
60 GB RAM 
1600 GB disk space SSD
Webmin + LAMP1 
1 Gbit/s port
1x IP address (let me know if you need more)
OS choices:
Ubuntu 16.04,18.04 or 20.04
Or Cent 7 or 8
or Debian 9 0r 10

**Comment by jhou5** [comment]
Server config doesn't help. To be helpful, I need to know how to reproduce the issue on a clean linux OS.

**Comment by Mediacast-cloud** [comment]
@jhou5 I prepared a clean ubuntu 18.04 OS. 
Will this suffice for you to reproduce the issue?

**Comment by jhou5** [comment]
i have clean ubuntu 18.04 OS installed, I don't know how to reproduce your issue.

**Comment by Mediacast-cloud** [comment]
It would help us better if you would carryout a new installation of the stack

**Comment by Mediacast-cloud** [comment]
Hi 
Still looking for help on OVC Stack installation.  payment can be made for this

**Comment by xwu2git** [comment]
Let's reset and sort out things a bit cleaner. 
Cloud you list explicitly what's the current blocking issues? Please be specific about what images are the issues.
I see the OWT image build is one of them and it is still not resolved? @qwu16

**Comment by Mediacast-cloud** [comment]
Thank you @xwu2git for the initiative. 
I hired a freelancer to carryout the installation of the 3 repos;
1- SVT-HEVC. 
2- WebRTC using Open RTC Kit “OWT’.
3- CDN Transcoding Sample.

The installer faced node dependency issues with the WebRTC ''OWT' and the CDN repos installations on an Ubuntu 18.04 OS from the GitHub branch master. He has since gave up.
First error was:
scripts/build.js -t all --check
while he ran this command
rtcConn.target.mk:199: recipe for target 'Release/obj.target/rtcConn/erizo/src/erizo/dtls/DtlsSocket.o'; failed
make: *** [Release/obj.target/rtcConn/erizo/src/erizo/dtls/DtlsSocket.o] Error 1
make: Leaving directory '/home/ubuntu/owt-server/source/agent/webrtc/rtcConn/build'
despite changing to the owt.m4 template after the fix of the node-gyp version to 6.1.0 and using the node 8.15.0 to build OWT

**Comment by qwu16** [comment]
It seems that your webrtc dependency is not successfully installed. Is there any network limit for your network to access Google website, WebRTC dependency needs to access to Google website to get some files. I am not sure which OWT source code version are you using, if you are not using the latest master branch OWT code, you get get the OWT dev image of 4.3.x which includes all the dependencies and 4.3.x OWT source code by: docker pull openvisualcloud/xeon-ubuntu1804-service-dev. With this image, you can modify OWT source code and build your own package.

**Comment by Mediacast-cloud** [comment]
Thank you @qwu16 
There are no network limits. The server side set up is at a Data Centre so access to a Google website is not a problem.
I am not sure which OWT version the freelance installer used but in the new installation efforts pulling docker files is a better approach to go for.

Do you know of a qualified installer who can carry out the stack installation for us? if not what basic skills should a freelancer have to be able to carry these installations?

**Comment by xwu2git** [comment]
I just built the following two builds from the master branch on a clean system:
(1) Dockerfiles: xeon-ubuntu1804-service-owt
(2) Video conferencing sample.
There is no build issue in either case.

Could you please retry? Make sure the proxy setting is correctly set if your are behind a firewall. The build system datetime must be set correctly. In general though, please build from a released branch. The latest is v20.7. Master might have temporarily hiccups due to active development.

**Comment by Mediacast-cloud** [comment]
Thank you @xwu2git for the efforts
I will have to hire a freelancer to carry out this installation. I will off course convey your recommendations
